core/templates/dev/head/pages/exploration_editor/exploration_metadata_modal_directive.html

```diff
@@ -10,15 +10,10 @@
                             label-text="Add a title: what should your exploration be called?"
                             form-style="{'font-weight':'normal'}">
   </exploration-title-editor>
-  <form name="objectiveForm">
-    <p ng-if="requireObjectiveToBeSpecified"
-       ng-class="{'has-error': (objectiveForm.objectiveInput.$touched || objectiveHasBeenPreviouslyEdited) && explorationObjectiveService.displayed.length < 15}">
```

Comment:
Would it make sense to also keep track of some of this stuff in the new directive's controller, so that we don't display an error before the user has even started typing?